src/Core/Settings.cpp

```diff
@@ -6863,6 +6863,19 @@ Both database and table names have to be unquoted - only simple identifiers are
 )", 0) \
     DECLARE(Bool, allow_general_join_planning, true, R"(
 Allows a more general join planning algorithm that can handle more complex conditions, but only works with hash join. If hash join is not enabled, then the usual join planning algorithm is used regardless of the value of this setting.
+)", 0) \
+    DECLARE(ObjectStorageGranularityLevel, cluster_table_function_split_granularity, ObjectStorageGranularityLevel::FILE, R"(
+Controls how data is split into tasks when executing a CLUSTER TABLE FUNCTION.
+
+This setting defines the granularity of work distribution across the cluster:
+- `file` — each task processes an entire file.
+- `row_group` — tasks are created per internal data block within a file (for example, Parquet row groups).
+
+Choosing finer granularity (like `row_group`) can improve parallelism when working with a small number of large files.
+For instance, if a Parquet file contains multiple row groups, enabling `row_group` granularity allows each group to be processed independently by different workers.
+)", 0) \
+    DECLARE(UInt64, cluster_table_function_buckets_batch_size, 0, R"(
+Approximate size of batch in distributed processing.
```

Comment:
Please, write more detailed explanation of this setting and the unit size (is it bytes size of the batch or number of buckets in the batch)